### ISSUE TITLE
Returning Json expects both 2 and 1 parameters simultaneously

### ISSUE THREAD

**Issue by coriolinus** [SergioBenitez/Rocket#470]
# Problem Statement

I'm trying to return a json object from a REST API I'm writing. The view I'm using looks, in full, like this:

```rust
/// View with which to create a user
#[post("/users", format = "application/json", data = "<user_data>")]
fn create_user(user_data: Json<UserData>, db: DB) -> Status<Json<Value>> {
    let conn = db.conn();
    // https://stackoverflow.com/questions/46905070/
    let user = or_return!(user_data.into_inner().into_user(&conn), |e| e);
    let uo = UserOutput::from(user);
    status!(
        Created,
        Json(uo)
    )
}
```

From the docs, this should probably work, but it produces the following error:

```shell
$ cargo check
error[E0061]: this function takes 2 parameters but 1 parameter was supplied
   --> src/views/user_account.rs:114:9
    |
114 |         Json(uo)
    |         ^^^^^^^^ expected 2 parameters

error: aborting due to previous error
```

# My Attempts to Debug

I tried adding a second parameter in the hopes that the typechecker would help me out, and things got more baffling: it now wants both 2 and 1 parameters simultaneously:

```shell
$ cargo check
error[E0061]: this function takes 2 parameters but 1 parameter was supplied
   --> src/views/user_account.rs:114:9
    |
114 |         Json(uo, 0)
    |         ^^^^^^^^^^^ expected 2 parameters

error[E0061]: this function takes 1 parameter but 2 parameters were supplied
   --> src/views/user_account.rs:114:14
    |
114 |         Json(uo, 0)
    |              ^^^^^ expected 1 parameter

error: aborting due to 2 previous errors
```

# Other Information

I wrote the `status!` macro myself to return arbitrary status codes; I'm pretty sure it's not causing the error here. Still, here's the expanded output for the relevant function:

```shell
$ cargo expand 2>/dev/null | grep "fn create_user" --before-context=1 --after-context=10
        #[rocket_route(static_rocket_route_info_for_create_user)]
        fn create_user(user_data: Json<UserData>, db: DB)
         -> Status<Json<Value>> {
            let conn = db.conn();
            let user =
                match user_data.into_inner().into_user(&conn) {
                    Ok(v) => v,
                    Err(e) => return (|e| e)(e),
                };
            let uo = UserOutput::from(user);
            ::status::Status::Created(::status::Created(Json(uo)))
        }
```

For reference there, `::status::Status` is a big enum of all legal HTTP return codes, which implements `Responder`; `::status::Created` is just a `pub use` of `rocket::response::status::Created`.

Rocket version, from `Cargo.toml`:

```toml
rocket = "0.3.3"
rocket_codegen = "0.3.3"
rocket_contrib = { version = "*", default-features = false, features = ["json"]}
```

OS: WSL on Windows 10

```shell
$ uname -a
Linux robocomp 4.4.0-43-Microsoft #1-Microsoft Wed Dec 31 14:42:53 PST 2014 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by coriolinus** [comment]
Nevermind, figured it out: `rocket::response::status::Created` is what expects two parameters, not `Json()`. Now I need to figure out how to generate a URI in a portable way (I'm pretty sure there's some kind of `uri!` macro, but it's un-searchable). Still, this isn't actually a Rocket bug.

**Comment by SergioBenitez** [comment]
The `uri!` macro is only available on the master branch and not yet available in any released version. It will be available in the next major release: v0.4.
